packages/components/src/tools-panel/tools-panel/hook.ts

```diff
@@ -81,44 +97,55 @@ export function useToolsPanel( props ) {
 	};
 
 	// Manage and share display state of menu items representing child controls.
-	const [ menuItems, setMenuItems ] = useState( {} );
+	const [ menuItems, setMenuItems ] = useState< ToolsPanelMenuItems >( {
+		default: {},
+		optional: {},
+	} );
 
 	const getResetAllFilters = () => {
-		const filters = [];
+		const filters: Array< Function > = [];
 
-		panelItems.forEach( ( item ) => {
+		panelItems.forEach( ( item: ToolsPanelItem ) => {
 			if ( item.resetAllFilter ) {
 				filters.push( item.resetAllFilter );
 			}
 		} );
-
 		return filters;
 	};
 
 	// Setup menuItems state as panel items register themselves.
 	useEffect( () => {
-		const items = generateMenuItems( { panelItems, reset: false } );
+		const items: ToolsPanelMenuItems = generateMenuItems( {
```

Comment:
TypeScript already knows that `generateMenuItems` returns `ToolsPanelMenuItems`

```suggestion
		const items = generateMenuItems( {
```